src/js/components/Util/CustomTable.tsx

```diff
@@ -0,0 +1,121 @@
+import { useEffect, useCallback, useMemo } from 'react';
+import { Table } from 'antd';
+import type { TableColumnType } from 'antd';
+import { useSearchParams } from 'react-router-dom';
+import { useTranslatedTableColumnTitles } from '@/hooks/useTranslatedTableColumnTitles';
+import { useNotify } from '@/hooks/notifications';
+import { useTranslationFn } from '@/hooks';
+
+import type { WithVisible } from '@/types/util';
+import { EXPANDED_QUERY_PARAM_KEY } from '@/constants/table';
+
+export interface CustomTableColumn<T> extends TableColumnType<T> {
+  isEmpty?: (value: any, record?: T) => boolean; // eslint-disable-line @typescript-eslint/no-explicit-any
+  isEmptyDefaultCheck?: boolean;
+}
+
+export type CustomTableColumns<T> = CustomTableColumn<T>[];
+
+type VisibilityFn<T> = (record: T) => boolean;
+type RowKeyFn<T> = (record: T, index?: number) => string;
+
+function addVisibility<T>(items: T[], isVisible: VisibilityFn<T>): WithVisible<T>[] {
+  return items.map((item) => ({ ...item, isVisible: isVisible(item) }));
+}
+
+function serializeExpandedKeys(keys: string[]) {
```

Comment:
make all of these functions consistent with the rest of bento public as usual – arrow functions.